tests/PKSim.Tests/Infrastructure/SnapshotSerializerSpecs.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using OSPSuite.BDDHelper;
+using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Utility;
+using PKSim.Core.Snapshots;
+using PKSim.Core.Snapshots.Services;
+using PKSim.Infrastructure.Serialization.Json;
+
+namespace PKSim.Infrastructure
+{
+   public abstract class concern_for_SnapshotSerializer : ContextSpecificationAsync<ISnapshotSerializer>
+   {
+      protected Parameter _parameter;
+      protected string _fileName;
+      protected Parameter _deserialiedParameter;
+
+      public override async Task GlobalContext()
+      {
+         await base.GlobalContext();
+         _fileName = FileHelper.GenerateTemporaryFileName();
+      }
+
+      protected override Task Context()
+      {
+         _parameter = new Parameter();
+         sut = new SnapshotSerializer();
+         return _completed;
+      }
+
+      public override async Task GlobalCleanup()
+      {
+         await base.GlobalCleanup();
+         FileHelper.DeleteFile(_fileName);
+      }
+   }
+
+  
+   public class When_serializing_a_parameter_with_a_value_set: concern_for_SnapshotSerializer
+   {
+      [TestCase(null)]
+      [TestCase(1.123456789)]
```

Comment:
Much easier to understand what's happening instead of trying to find names that make sense :)